src/Misc/layoutroot/run.sh

```diff
@@ -11,54 +11,60 @@ fi
 # https://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within
 SOURCE="${BASH_SOURCE[0]}"
 while [ -h "$SOURCE" ]; do # resolve $SOURCE until the file is no longer a symlink
-  DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
-  SOURCE="$(readlink "$SOURCE")"
-  [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
+    DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
+    SOURCE="$(readlink "$SOURCE")"
+    [[ $SOURCE != /* ]] && SOURCE="$DIR/$SOURCE" # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
 done
 DIR="$( cd -P "$( dirname "$SOURCE" )" && pwd )"
 
 # Do not "cd $DIR". For localRun, the current directory is expected to be the repo location on disk.
 
 # Run
 shopt -s nocasematch
-if [[ "$1" == "localRun" ]]; then
-    "$DIR"/bin/Runner.Listener $*
-else
-    "$DIR"/bin/Runner.Listener run $*
 
-# Return code 3 means the run once runner received an update message.
-# Sleep 5 seconds to wait for the update process finish
-    returnCode=$?
-    if [[ $returnCode == 3 ]]; then
-        if [ ! -x "$(command -v sleep)" ]; then
-            if [ ! -x "$(command -v ping)" ]; then
-                COUNT="0"
-                while [[ $COUNT != 5000 ]]; do
-                    echo "SLEEP" > /dev/null
-                    COUNT=$[$COUNT+1]
-                done
-            else
-                ping -c 5 127.0.0.1 > /dev/null
-            fi
-        else
-            sleep 5
-        fi
-    elif [[ $returnCode == 4 ]]; then
-        if [ ! -x "$(command -v sleep)" ]; then
-            if [ ! -x "$(command -v ping)" ]; then
-                COUNT="0"
-                while [[ $COUNT != 5000 ]]; do
-                    echo "SLEEP" > /dev/null
-                    COUNT=$[$COUNT+1]
-                done
-            else
-                ping -c 5 127.0.0.1 > /dev/null
-            fi
+safe_sleep() {
+    if [ ! -x "$(command -v sleep)" ]; then
+        if [ ! -x "$(command -v ping)" ]; then
+            COUNT="0"
+            while [[ $COUNT != 5000 ]]; do
```

Comment:
i thought the update script waits longer than 5 seconds... or do we not reach this point until after the update script finishes?

Reply:
discussed offline, this is so the update script can perform the folder move, etc

Reply:
...does that mean run.cmd is still running? wondering if that's a problem since we're moving the folder... thinking...

ok i think windows will let us move a folder even if a file is still in use, so i guess cmd.exe has a handle open to the file and it shouldnt disrupt the execution of the script. might want to test that theory by running a simple cmd script in a loop and moving the folder containing the script.

Reply:
i wonder if we *could* service it by moving it. still might be an interesting test:

running a simple cmd script that executes a loop and move the folder containing the script

Reply:
Modifying `run.cmd` while it's running is super unpredictable as it is being read byte by byte afaik by `cmd.exe`.
If we added a new line, it could be before or after the pointer, if we removed it, `cmd.exe` would see it as 'eof' and terminate `run.cmd`.